Technical-Workshop/Presentation/Home Screen/View/HomeScreenViewController.swift

```diff
@@ -10,46 +10,80 @@ import UIKit
-        let cell = categoriesCollectionView.dequeueReusableCell(withReuseIdentifier: "collectionViewCell", for: indexPath) as! CategoryCollectionViewCell
-        let currentCategoryItem = viewModel.categoryItem(at: indexPath.row)
-        cell.configure(categoryItem: currentCategoryItem)
-        return cell
+        
+        if collectionView == self.categoriesCollectionView {
+            let cell = categoriesCollectionView.dequeueReusableCell(withReuseIdentifier: "categoriesCollectionViewCell", for: indexPath) as! CategoryCollectionViewCell
+            let currentCategoryItem = viewModel.categoryItem(at: indexPath.row)
+            cell.configure(categoryItem: currentCategoryItem)
+            return cell
+        } else{
+            let cell = mealsCollectionView.dequeueReusableCell(withReuseIdentifier: "MealCollectionViewCell", for: indexPath) as! MealCollectionViewCell
+            let currentMealItem = viewModel.mealItem(at: indexPath.row)
+            cell.configure(mealItem: currentMealItem)
+            return cell
+        }
     }
+ 
+    // MARK: Collection View: setting sizes of items
     
     func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
-        let width = categoriesCollectionView.bounds.size.width * 0.2
-        let height = categoriesCollectionView.bounds.size.height
-        return CGSize(width: width, height: height)
-    }
-}
-
-extension HomeScreenViewController: UITableViewDelegate, UITableViewDataSource {
-    // MARK: Table View Callback Methods
-
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        0
+        if collectionView == self.categoriesCollectionView {
+            let width = categoriesCollectionView.bounds.size.width * 0.2
+            let height = categoriesCollectionView.bounds.size.height
+            return CGSize(width: width, height: height)
+        } else{
+            let width = mealsCollectionView.bounds.size.width
+            let height = mealsCollectionView.bounds.size.height * 0.4
+            return CGSize(width: width, height: height)
+        }
     }
+  
+    // MARK: Collection View: Selecting an item
     
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        UITableViewCell()
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        if collectionView == self.categoriesCollectionView {
+            let cells = collectionView.visibleCells
+            for cell in cells {
+                guard let cell = cell as? CategoryCollectionViewCell else { return }
+                if cell.isSelected {
+                    cell.categoryImageView.backgroundColor = UIColor(named: "categoriesBackgroundColorSelected")
```

Comment:
we can discuss how we can move the logic of changing the background color into the cell